src/axom/core/utilities/StringUtilities.hpp

```diff
@@ -174,6 +174,23 @@ std::string removeBeforeDelimiter(const std::string& str, const char delim = '/'
 std::string removeAllInstances(const std::string& target,
                                const std::string& substr);
 
+/*!
+*****************************************************************************
+* \brief This function finds all instances of a substring and replaces it
+* with another string for the target string
+*
+* \param [in] target The string to operate on
```

Comment:
Should this argument be `inout`?

Reply:
In the example from [cppreference](https://en.cppreference.com/w/cpp/string/basic_string/replace) @kennyweiss gave it is `inout`, but I have it as an `[in]` here to be consistent with the other functions in `StringUtilities`.